src/Interpreters/ExpressionActions.cpp

```diff
@@ -157,9 +158,11 @@ static void setLazyExecutionInfo(
 
     const ActionsDAGReverseInfo::NodeInfo & node_info = reverse_info.nodes_info[reverse_info.reverse_index.at(node)];
 
+    std::cout << "xxx Node info uir " << node_info.used_in_result << ", parents: " << node_info.parents.size() << ", type " << ActionsDAG::typeToString(node->type) << ", result name:" << node->result_name << std::endl;
     /// If node is used in result or it doesn't have parents, we can't enable lazy execution.
-    if (node_info.used_in_result || node_info.parents.empty() || (node->type != ActionsDAG::ActionType::FUNCTION && node->type != ActionsDAG::ActionType::ALIAS))
+    if (node_info.parents.empty() || (node->type != ActionsDAG::ActionType::FUNCTION && node->type != ActionsDAG::ActionType::ALIAS))
```

Comment:
It won't work. We cannot execute expression lazily if it's used in query result, because we will need full result of this expression anyway.
Simple example:
```
SELECT expression, condition1 OR some_func(expression) FROM ...
```
In this case we cannot execute `expression` lazily.